workspaces/shorturl/plugins/shorturl/src/components/ShortURLPage/ShortURLCreate.tsx

```diff
@@ -0,0 +1,160 @@
+// code based on https://github.com/shailahir/backstage-plugin-shorturl
+import { Link } from '@backstage/core-components';
+import {
+  Box,
+  Button,
+  Card,
+  CardContent,
+  Grid,
+  TextField,
+} from '@material-ui/core';
+import React, { useEffect, useState } from 'react';
+import {
+  useApi,
+  alertApiRef,
+  discoveryApiRef,
+  configApiRef,
+  fetchApiRef,
+  identityApiRef,
+} from '@backstage/core-plugin-api';
+import { DefaultShortURLApi } from '../../api';
+import useAsync from 'react-use/lib/useAsync';
+
+interface ShortURLCreateProps {
+  onCreate: () => void;
+}
+
+export const ShortURLCreate: React.FC<ShortURLCreateProps> = ({ onCreate }) => {
+  const alertApi = useApi(alertApiRef);
+  const fetchApi = useApi(fetchApiRef);
+  const configApi = useApi(configApiRef);
+  const identityApi = useApi(identityApiRef);
+  const discoveryApi = useApi(discoveryApiRef);
+  const shortURLApi = new DefaultShortURLApi(
+    fetchApi,
+    discoveryApi,
+    identityApi,
+  );
+
+  const [longUrl, setLongUrl] = useState('');
+  const [shortUrl, setShortUrl] = useState('');
+
+  const { value: baseUrl } = useAsync(async () => {
+    return await configApi.getString('app.baseUrl');
+  }, []);
+
+  const validateUrl = (url: string): boolean => {
+    try {
+      const validUrl = new URL(url);
+      return !!validUrl;
+    } catch (_) {
+      return false;
+    }
+  };
```

Comment:
```suggestion
function isValidUrl(url: string): boolean {
  try {
    // eslint-disable-next-line no-new
    new URL(url);
    return true;
  } catch {
    return false;
  }
}
```

I'm surprised we don't have a common package for URL validation. That said, this is how we handle URL validation in the backend, and it should work here as well. Additionally, you might want to avoid defining functions within React components, as it can negatively impact React's performance.